hypothesis-python/src/hypothesis/core.py

```diff
@@ -892,8 +893,11 @@ def fake_subTest(self, msg=None, **__):
     yield
 
 
-def given(*given_arguments, **given_kwargs):
-    # type: (*SearchStrategy, **SearchStrategy) -> Callable
+def given(
+    *given_arguments,  # type: Union[SearchStrategy, UniqueIdentifier]
```

Comment:
Doesn't need to be as part of this PR, but it would be nice to tighten up these types to be more specific about *which* `UniqueIdentifier` is permitted here, if only so the `UniqueIdentifier` type doesn't appear in our public APIs.